src/Core/Environment.php

```diff
@@ -89,17 +89,20 @@ public static function increaseMemoryLimitTo($memoryLimit = -1)
             return true;
         }
 
-        // Check hard maximums
-        $max = static::getMemoryLimitMax();
-        if ($max > 0 && ($memoryLimit < 0 || $memoryLimit > $max)) {
-            $memoryLimit = $max;
-        }
 
         // Increase the memory limit if it's too low
         if ($memoryLimit < 0) {
             ini_set('memory_limit', '-1');
-        } elseif ($memoryLimit > $curLimit) {
-            ini_set('memory_limit', Convert::bytes2memstring($memoryLimit));
+        } else {
+            // Check hard maximums
+            $max = static::getMemoryLimitMax();
```

Comment:
The method behaviour is now inconsistent.
We'll either need to check `max` above for `$memoryLimit < 0` too, or do not check it at all.

Currently the patch makes it ignore the max limit for `unlimited`, but respect for particular numbers.